rosbag2_transport/test/rosbag2_transport/test_record_services.cpp

```diff
@@ -226,3 +247,37 @@ TEST_F(RecordSrvsTest, pause_resume)
   is_paused_response = successful_service_request<IsPaused>(cli_is_paused_);
   EXPECT_FALSE(is_paused_response->paused);
 }
+
+TEST_F(RecordSrvsTest, stop_start_discovery)
+{
+  EXPECT_FALSE(recorder_->is_discovery_stopped());
+  auto is_discovery_stopped_response =
+    successful_service_request<IsDiscoveryStopped>(cli_is_discovery_stopped_);
+  EXPECT_FALSE(is_discovery_stopped_response->stopped);
+
+  successful_service_request<StopDiscovery>(cli_stop_discovery_);
+  EXPECT_TRUE(recorder_->is_discovery_stopped());
+  is_discovery_stopped_response =
+    successful_service_request<IsDiscoveryStopped>(cli_is_discovery_stopped_);
+  EXPECT_TRUE(is_discovery_stopped_response->stopped);
+
+  successful_service_request<StartDiscovery>(cli_start_discovery_);
+  EXPECT_FALSE(recorder_->is_discovery_stopped());
+  is_discovery_stopped_response =
+    successful_service_request<IsDiscoveryStopped>(cli_is_discovery_stopped_);
+  EXPECT_FALSE(is_discovery_stopped_response->stopped);
+}
+
+TEST_F(RecordSrvsTest, record_stop)
+{
+  auto & writer = recorder_->get_writer_handle();
+  auto & mock_writer = dynamic_cast<MockSequentialWriter &>(writer.get_implementation_handle());
+
+  EXPECT_FALSE(mock_writer.closed_was_called());
+
+  successful_service_request<Stop>(cli_stop_);
+  EXPECT_TRUE(mock_writer.closed_was_called());
+
+  successful_service_request<Record>(cli_record_);
+  EXPECT_FALSE(mock_writer.closed_was_called());
+}
```

Comment:
Please add recorder start after the stop and then again stop to make sure that we can start again after the stop